cli/assets/templates/wat/main.wat

```diff
@@ -0,0 +1,99 @@
+;;
+;; WASM-4: https://wasm4.org/docs
+
+(import "env" "memory" (memory 1))
+
+;; ┌───────────────────────────────────────────────────────────────────────────┐
+;; │                                                                           │
+;; │ Drawing Functions                                                         │
+;; │                                                                           │
+;; └───────────────────────────────────────────────────────────────────────────┘
+(; Copies pixels to the framebuffer. ;)
+(import "env" "blit" (func $blit (param i32 i32 i32 i32 i32 i32)))
+
+(; Copies a subregion within a larger sprite atlas to the framebuffer. ;)
+(import "env" "blitSub" (func $blitSub (param i32 i32 i32 i32 i32 i32 i32 i32 i32)))
+
+(; Draws a line between two points. ;)
+(import "env" "line" (func $line (param i32 i32 i32 i32)))
+
+(; Draws a horizontal line. ;)
+(import "env" "hline" (func $hline (param i32 i32 i32)))
+
+(; Draws a vertical line. ;)
+(import "env" "vline" (func $vline (param i32 i32 i32)))
+
+(; Draws an oval (or circle). ;)
+(import "env" "oval" (func $oval (param i32 i32 i32 i32)))
+
+(; Draws a rectangle. ;)
+(import "env" "rect" (func $rect (param i32 i32 i32 i32)))
+
+(; Draws text using the built-in system font. ;)
+(import "env" "text" (func $text (param i32 i32 i32)))
+
+;; ┌───────────────────────────────────────────────────────────────────────────┐
+;; │                                                                           │
+;; │ Sound Functions                                                           │
+;; │                                                                           │
+;; └───────────────────────────────────────────────────────────────────────────┘
+(; Plays a sound tone. ;)
+(import "env" "tone" (func $tone (param i32 i32 i32 i32)))
+
+;; ┌───────────────────────────────────────────────────────────────────────────┐
+;; │                                                                           │
+;; │ Storage Functions                                                         │
+;; │                                                                           │
+;; └───────────────────────────────────────────────────────────────────────────┘
+(; Reads up to `size` bytes from persistent storage into the pointer `dest`. ;)
+(import "env" "diskr" (func $diskr (param i32 i32)))
+
+(; Writes up to `size` bytes from the pointer `src` into persistent storage. ;)
+(import "env" "diskw" (func $diskw (param i32 i32)))
+
+(; Prints a message to the debug console. ;)
+(import "env" "trace" (func $trace (param i32)))
+
+(; Prints a message to the debug console. ;)
+(import "env" "tracef" (func $tracef (param i32 i32)))
+
+(data (i32.const 0x19a0)
+  ;; smiley @ 0x19a0
+  "\c3\81\24\24\00\24\99\c3"
+
+  ;; 0x19a8
```

Comment:
It definitely is clearer. But it doesn't prevent overlaps, since the data segments are allowed to overlap. And it does make the final `.wasm` file size slightly larger. But I think you're right that the clarity would be worth it for the guide.